### DISCUSSION TITLE
What is it?

### DISCUSSION THREAD

**Discussion by Seanmclem** [johnlindquist/kit#56] [Q&A] [answered]
I'm finding the description "automate anything" to be a little brief and vague. Maybe the readme could have some more info.

**Comment by johnlindquist** [comment, marked as answer]
Here's a quick run-down off the top of my head:

## Goals of Script Kit
Make scripts:
1. Easier to write
2. Easier to run
3. Easier to share
4. Future work - easier to export to other services

## Easy to write
* Script Kit scripts are written in JavaScript
* Create a new script by typing the name and hitting Enter!
* Script Kit launches your editor/enviornment of choices
* Script Kit leverages the npm ecosystem, so you have a world of utilities at your fingertips
* Script Kit ships with many utilities for common tasks with windows, files, text, etc

## Easy to run
* Scripts run from keyboard shortcuts
* Add a shortcut to a script with a comment, e.g., `//Shortcut: opt + j`
* Scripts also run from the terminal `~/.kenv/bin/hello-world`
* Scripts prompt for missing arguments (in both the App and the terminal)
* Provide arguments in the terminal `~/.kenv/bin/hello-world John`
* Provide arguments for the app `await run("hello-world", "John")`
* You can run scripts from any of your favorite tools: Karabiner, Better Touch Tool, Keyboard Maestro, or even Siri shortcuts from your phone

## Easy to share
* Cross platform scripts! Mac/Window/Linux are all welcome
* Each script has a "share as link" option which creates a gist of your script and generates a link [like this](kit://color-convert?url=https://scriptkit.app/scripts/johnlindquist/color-convert.js) to "install" your script
* A common language in JavaScript makes it simple to tweak someone else's script
* It's encouraged duplicate scripts for variations of tasks
* There's no "config", only conventions. So everyone is using the same system.
* npm packages make sharing super complex tasks very simple

## Easier to Export
* A big focus of 2022 is adding an "Export to" where script can be push used as GitHub Actions, serverless functions, or anywhere else that a "single file" is used to create a task

**Reply by christiangenco** [reply]
Oh! It's like [Alfred](https://www.alfredapp.com/) but way easier to program, or [Hammerspoon](https://www.hammerspoon.org/) but with standard ways of triggering shortcuts and it's javascript instead of Lua.

Is that right?

**Reply by johnlindquist** [reply]
Here's how I look at it:

Script Kit started as a terminal-only project where I was experimenting with easier ways to create/run/share scripts. The app is a UI wrapper around that project. Script Kit still works in terminal-only mode (which I use _a lot_), so I don't really think about it in comparison to other apps.

**Comment by mandarvaze** [comment]
I've created a PR (https://github.com/johnlindquist/kit/pull/90) based on the above. I hope it is helpful.

**Reply by johnlindquist** [reply]
Thanks for putting that together. Since Script Kit is in beta and I expect to make a couple breaking changes before final release, I don't really want to put together any sort of docs/materials that imply that it's "ready". I'm happy there are so many people willing to dive-in (~250 downloads as of today) before docs are ready and I'll do my best to support them, but I don't want a bunch of people angry at me when I change something 😬

I promise we'll have solid explanations/docs/examples/etc once everything settles for a final release.
